src/reminder/reminder.service.ts

```diff
@@ -192,19 +217,20 @@ export class ReminderService {
     return recurrings;
   }
 
-  async deleteReminder(rid: number): Promise<string> {
-    const reminder = await this.reminderRepository.findOne({ where: { rid: rid } });
+  async deleteReminder(rid: number, uid: number): Promise<string> {
+    const reminder = await this.reminderRepository.findOne({ where: { rid: rid, uid: uid } });
     if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND);
-
     await this.reminderRepository.delete(reminder);
+    this.clearReminderJobs(reminder.rid);
     return 'Complete';
   }
 
-  async getReminder(rid: number): Promise<GetReminder> {
-    const reminder = await this.reminderRepository.findOne({ where: { rid: rid } });
+  async getReminder(rid: number, uid: number): Promise<ReminderDto> {
+    const reminder = await this.reminderRepository.findOne({ where: { rid: rid, uid: uid }, relations: ['recurring'] });
     if (!reminder) throw new HttpException('Reminder not found', HttpStatus.NOT_FOUND);
-
-    return reminder;
+    delete reminder.recurrings;
+    const schedule = this.getScheduleFromRecurring(reminder.rid, reminder.recurrings, reminder.startingDateTime);
```

Comment:
await?

Reply:
`getScheduleFromRecurring` does not return a promise